src/lib.rs

```diff
@@ -276,6 +301,26 @@ pub fn encode_upper<T: AsRef<[u8]>>(data: T) -> String {
     data.encode_hex_upper()
 }
 
+/// Encodes `data` as hex string using uppercase characters, appending to target string.
+///
+/// This is the same as [`encode_to`], but uses uppercase characters.
+///
+/// # Example
+///
+/// ```
+/// let mut s = "The hex encoding is: ".to_string();
+/// hex::encode_to_upper("Hello world!", &mut s);
+/// assert_eq!(s, "The hex encoding is: 48656C6C6F20776F726C6421");
+/// ```
+#[must_use]
+#[cfg(feature = "alloc")]
+pub fn encode_to_upper<T: AsRef<[u8]>>(data: T, s: &mut String) {
```

Comment:
Or hm, maybe this should be `encode_upper_to`, since it is confusing with `encode_to_slice`, we're not encoding to an upper :smile: